packages/@aws-cdk/aws-stepfunctions/lib/fields.ts

```diff
@@ -0,0 +1,136 @@
+import { findReferencedPaths, JsonPathToken, renderObject } from "./json-path";
+
+/**
+ * Extract a field from the State Machine data that gets passed around between states
+ */
+export class DataField {
+  /**
+   * Instead of using a literal string, get the value from a JSON path
+   */
+  public static fromStringAt(path: string): string {
+    validateDataPath(path);
+    return new JsonPathToken(path).toString();
+  }
+
+  /**
+   * Instead of using a literal string list, get the value from a JSON path
+   */
+  public static fromListAt(path: string): string[] {
+    validateDataPath(path);
+    return new JsonPathToken(path).toList();
+  }
+
+  /**
+   * Instead of using a literal number, get the value from a JSON path
+   */
+  public static fromNumberAt(path: string): number {
+    validateDataPath(path);
+    return new JsonPathToken(path).toNumber();
+  }
+
+  /**
+   * Use the entire data structure
+   *
+   * Will be an object at invocation time, but is represented in the CDK
+   * application as a string.
+   */
+  public static get entirePayload(): string {
+    return new JsonPathToken('$').toString();
+  }
+
+  private constructor() {
+  }
+}
+
+/**
+ * Extract a field from the State Machine Context data
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#wait-token-contextobject
+ */
+export class ContextField {
+  /**
+   * Instead of using a literal string, get the value from a JSON path
+   */
+  public static fromStringAt(path: string): string {
+    validateContextPath(path);
+    return new JsonPathToken(path).toString();
+  }
+
+  /**
+   * Instead of using a literal number, get the value from a JSON path
+   */
+  public static fromNumberAt(path: string): number {
+    validateContextPath(path);
+    return new JsonPathToken(path).toNumber();
+  }
+
+  /**
+   * Return the Task Token field
+   *
+   * External actions will need this token to report step completion
+   * back to StepFunctions using the `SendTaskSuccess` or `SendTaskFailure`
+   * calls.
+   */
+  public static get taskToken(): string {
+    return new JsonPathToken('$$.Task.Token').toString();
+  }
+
+  /**
+   * Use the entire context data structure
+   *
+   * Will be an object at invocation time, but is represented in the CDK
+   * application as a string.
+   */
+  public static get entireContext(): string {
+    return new JsonPathToken('$$').toString();
+  }
+
+  private constructor() {
+  }
+}
+
+/**gg
+ * Helper functions to work with structures containing fields
+ */
+export class FieldUtils {
```

Comment:
Do we need this public?

Reply:
Yes, it is used in the `-tasks` library.